src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Android/AndroidSdkResolver.cs

```diff
@@ -60,6 +61,30 @@ public static string GetJavaSdkPath ()
 			return JavaSdkPath;
 		}
 
+		static string JavaSdkVersionString;
+
+		public static string GetJavaSdkVersionString ()
+		{
+			if (string.IsNullOrEmpty (JavaSdkVersionString)) {
+				var javaPath = Path.Combine (GetJavaSdkPath (), "bin", "java");
+				var psi = new ProcessStartInfo (javaPath, "-version") {
+					CreateNoWindow = true,
+					RedirectStandardOutput = true,
+					RedirectStandardError = true,
+					WindowStyle = ProcessWindowStyle.Hidden,
+					UseShellExecute = false,
+				};
```

Comment:
I've seen `java -version` randomly take 2 seconds sometimes. Can we cache the result, so this would only be called once?

Reply:
The `JavaSdkVersionString` static field used here should prevent subsequent calls from having to invoke java